ts/parts/SvgRenderer.ts

```diff
@@ -2717,10 +2717,28 @@ extend((
             strokeWidth,
             shadowWidth,
             shadowElementOpacity,
+            update = false,
 
             // compensate for inverted plot area
             transform;
 
+        // Update shadow when options change (#12091).
+        if (isObject(shadowOptions)) {
+            if (this.oldShadowOptions) {
+                Object.keys(this.oldShadowOptions || {}).forEach((key): void => {
```

Comment:
We have `objectEach` method you can use here.

Reply:
Extras:
What if `oldShadowOptions = { width: 1, opacity: 1 }` and `newShadowOptions = { offsetX: 10, offsetY: 10 }` ? From code (a few `pick()` calls) it seems that might be the case.